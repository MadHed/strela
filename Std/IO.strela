// Copyright (c) 2018 Stephan Unverwerth
// This code is licensed under MIT license (See LICENSE for details)

module Std.IO {
    export function print(v: int): void {}
    export function print(v: float): void {}
    export function print(v: String): void {}
    export function print(v: bool): void {}

    export function println(v: int): void { print(v); print("\n"); }
    export function println(v: float): void { print(v); print("\n"); }
    export function println(v: String): void { print(v); print("\n"); }
    export function println(v: bool): void { print(v); print("\n"); }
}