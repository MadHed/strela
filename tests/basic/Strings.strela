module Strings {
    import Std.IO.println;

    function main(args: String[]): int {
        var empty = new String;
        println(empty.length());

        var four = new String(4);
        println(four.length());

        var fromarray = new String([65 as u8, 66 as u8, 67 as u8]);
        println(fromarray);

        var fromarrayslice = new String([65 as u8, 66 as u8, 67 as u8], 1);
        println(fromarrayslice);

        var frompair = new String("Hello", ", world!");
        println(frompair);

        println(fromarray.length());

        println("Hello" + " darkness");

        println("A" != "B");
        println("A" != "A");
        println("A" != "");

        println("A" == "B");
        println("A" == "A");
        println("A" == "");

        println("Quite".substr(0, 4));

        println("XYZ"[0]);
        println("XYZ"[1]);
        println("XYZ"[2]);

        var cities = "London,New York,Berlin";

        var parts = cities.split(",");
        println(parts.length);
        println(parts[1]);

        println(cities.split("###").length);
        println(cities.split("").length);

        println("  \t\r\n   Go   \n\r\n\t\r".trim());
		println("     ".trim());

        println("123456".toInt());
        println("123.456".toFloat());

        println("Hello, world".replace("e", "a"));

        return 0;
    }
}