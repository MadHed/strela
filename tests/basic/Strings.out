0
4
ABC
A
Hello, world!
3
Hello darkness
true
false
true
false
true
false
Quit
X
Y
Z
3
New York
1
1
Go

123456
123.456
Hallo, world
